openstack/client.go

```diff
@@ -16,6 +17,8 @@ const (
 	v30 = "v3.0"
 )
 
+var VERSION_PATTERN = regexp.MustCompile("/v[1-9]{1}[0-9.]*")
```

Comment:
You are right, the same functionality could be achieved with a simple if else structure. Another matter is, should the function return an error if a version other than "v2.0" or "v3" is found, or should it be stripped, or included as part of IdentityBase? My opinion is that we should check with regexp "/v[1-9]{1}[0-9.]{0,2}" and return an error if a match other than v2.0 or v3 is found. What you think?